crates/cli/src/lib.rs

```diff
@@ -542,21 +544,32 @@ pub fn test(matches: &ArgMatches, triple: Triple) -> io::Result<i32> {
 
     let mut writer = std::io::stdout();
 
-    let (failed_count, passed_count) = roc_repl_expect::run::run_toplevel_expects(
-        &mut writer,
-        roc_reporting::report::RenderTarget::ColorTerminal,
-        arena,
-        interns,
-        &layout_interner.into_global(),
-        &lib,
-        &mut expectations,
-        expects,
-    )
-    .unwrap();
+    let mut total_failed_count = 0;
+    let mut total_passed_count = 0;
+
+    let mut results_by_module = Vec::new();
+    let global_layout_interner = layout_interner.into_global();
+    for (module_id, expects) in expects_by_module.into_iter() {
+        let test_start_time = Instant::now();
```

Comment:
One thing to note here: previously the test duration we reported included the time taken by `load_and_monomorphize`, `expect_mono_module_to_dylib`, etc.; now the per-module time only includes the time it took to actually run the tests. I think that's the only way to sensibly split up the duration, but it means that we often report a value of 0ms: `1 failed and 22 passed in 0 ms.`

Reply:
I've asked for some extra input [on zulip](https://roc.zulipchat.com/#narrow/stream/395097-compiler-development/topic/test.20time.20reporting/near/426187328).